src/review/logic.py

```diff
@@ -192,21 +192,24 @@ def get_decision_content(request, article, decision, author_review_url):
     return render_template.get_message_content(request, email_context, template_name)
 
 
-def get_revision_request_content(request, article, revision):
-    do_revisions_url = request.journal.site_url(path=reverse(
-        'do_revisions',
-        kwargs={
-            'article_id': article.pk,
-            'revision_id': revision.pk,
-        }
-    ))
-
+def get_revision_request_content(request, article, revision, draft=False):
     email_context = {
         'article': article,
         'revision': revision,
-        'do_revisions_url': do_revisions_url,
     }
 
+    if not draft:
+        do_revisions_url = request.journal.site_url(path=reverse(
+            'do_revisions',
+            kwargs={
+                'article_id': article.pk,
+                'revision_id': revision.pk,
+            }
+        ))
+        email_context['do_revisions_url'] = do_revisions_url
+    else:
+        email_context['do_revisions_url'] = "{{ do_revisions_url }}"
```

Comment:
How does this work? presumably the template already has the variable `{{ do_revisions_url }}` . If we don't render the context variable here, how does it get rendered? Is it expecting to be processed by the event later?

Reply:
We retain the placeholder `{{ do_revisions_url }}` in the drafted template so it can be rendered later as the request does not exist at this point.